app/utils/watchdog.py

```diff
@@ -1,8 +1,7 @@
-from functools import partial
-import json
-from pathlib import Path
 import os
+from functools import partial
 from threading import Timer
+from typing import Optional
```

Comment:
Optional typing in python3.10+ is recommended to use the union op as Type | None instead of Optional[Type]

Reply:
I  just added the missing import i am not going to modify watchdog since i am sure i will break it hope twsta or you can look into it